### ISSUE TITLE
[Feat Req]: Table panel with clickable cell links using any row column value in link

### ISSUE THREAD

**Issue by qed-** [grafana/grafana#3754]
Similar to #3544 I use the table panel to track process up/down status.

It would be great if clicking on a table row could pass the value of a specified cell as a URL parameter to the drill down dashboard.

**Comment by costimuraru** [comment]
That would be awesome!

**Comment by tstm** [comment]
+1

**Comment by digrich** [comment]
+1

**Comment by futamint** [comment]
+1

**Comment by 3IMOH** [comment]
+1
It would be great if this could also be implemented within the "Column Styles". That way you could click on a specific value to drill down into a more detailed dashboard

**Comment by mnietz** [comment]
:+1:

**Comment by indranilr** [comment]
+1
This is feature is needed to not only drill down to another dashboard but to navigate to external link with values of the chosen values of the table cell values or bars/points in a bar/line graph  (As described in #1020)

**Comment by dmitrios** [comment]
+1
I hope this will be available in 4.0 at least for drill down from table cell to another dashboard. This real show stopper for hierarchical navigation with InfluxDb.

**Comment by gtt116** [comment]
+1

**Comment by jayannah** [comment]
Is this fixed in the latest release? I have also the same requirements and unable to achieve.

**Comment by dmitrios** [comment]
I submitted #6002 for this exact issue a while back.

**Comment by n0012** [comment]
+1

**Comment by mady4ever** [comment]
+1

**Comment by dmitrios** [comment]
@bergqist, Any chance that #6002 could be fixed soon?

**Comment by dmitrios** [comment]
Hi @bergquist, is there any chance for fixing #6002 relatively soon?

**Comment by MadeInChina** [comment]
+1

**Comment by inselbuch** [comment]
+1

**Comment by inselbuch** [comment]
Even better, clicking on a row should be able to a particular column for that row... even if it is hidden. Sometimes you have a key that you want to pass, but not display (because it is a GUID or some other cryptic thing).

**Comment by jonesn** [comment]
+1

**Comment by chrisjschultz** [comment]
+1

**Comment by daianmartinho** [comment]
+1

**Comment by rmleach** [comment]
👍 +1

Generating roll-up status tables that are clickable feels so much cleaner than clunky-feeling drown-down variables based on queries.

**Comment by dmitrios** [comment]
+1 I ended up implementing new column style for the link myself. Works perfectly for drill down using master-detail dashboards.

**Comment by inselbuch** [comment]
@dmitrios, are you making that available? can we download/test?

**Comment by alogan-nyansa** [comment]
@dmitrios I would like to test as well if you make it available.

**Comment by tammytankian** [comment]
+1

**Comment by dmitrios** [comment]
I posted sample code. It is very simple example and works fine with 4+ version including the latest.

**Comment by inselbuch** [comment]
Posted sample code where?

**Comment by daianmartinho** [comment]
dmitios, 
First of all, thank you for your time working on this.
I'm trying to install your plugin but i can't see it on plugins list.
I have moved reftable/ folder to /grafana/plugins/ and restarted the server without success.

**Comment by dmitrios** [comment]
@daianmartinho Please get the latest from my repository and follow the instructions in readme there to deploy plugin. You can always log issue to that repository if you will encounter further problems.

**Comment by AlysG** [comment]
@dmitrios Thanks for this plugin, I just need this feature. I installed your plugin but I failed to use it. Can you give us an example?
I'm using grafana 4.2.0. If I create a refTable and just add a column style for 1 column, this column disappears. Can you help me?
![image](https://cloud.githubusercontent.com/assets/28618869/25937209/0cff77c8-362b-11e7-8a7f-e6f07d352059.png)

**Comment by dmitrios** [comment]
@AlysG I could not reproduce what you describe. Feel free to create issue at https://github.com/dmitrios/grafana-plugins-reftable instead.

**Comment by Kenith** [comment]
I found a walk around for the case, with the case you could add the hyperlink for each cell.
The solution is kind of tricky.

1. Get the cell data from database, and make the data with **a** tag, for example, `<a href="www.google.com">Google</a>`;

2. Go to the Grafana path, and find the index.html, MAC example, `/usr/local/opt/grafana/share/grafana/public/views/index.html`;

3. Add the `<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.6.2/jquery.min.js"></script>` to the head of the index.html;

4. Add the following code to the bottom of the index.html
```
	<script type="text/javascript">
		setTimeout(function() {
			$('tbody > *  > td:nth-child(4)').each(function(){
			    	var $this = $(this);
			    	var t = $this.text();
			    	$this.html(t.replace('&lt;','<').replace('&gt;', '>'));
					});
				}, 10000);		
	</script>
```
5. Restart the grafana service, MAC as example, `brew services restart grafana`.

**Comment by alexanderzobnin** [comment]
@torkelo should we close this one because PR was merged?

**Comment by inselbuch** [comment]
I AM SO HOT FOR THIS FEATURE I AM SWEATING. PLEASE RELEASE 5.0.0 ASAP.

**Comment by torkelo** [comment]
A screenshot showing the options for this feature:

![image](https://user-images.githubusercontent.com/10999/28797744-a2c70f62-7642-11e7-8b40-3b502807d5b2.png)

To make a column with link cells you add a column style and check the `Render value as link` checkbox. That will make the link url & tooltip options appear. 

You can use the cell value or other cell values in the link url using `$__cell` or  `$__cell_n`' where n refers to Nth column value in current row. Column indexes are started from 0. For instance, $__cell_1 refers to second column's

**Comment by jonesn** [comment]
Hi Just some feedback on this. We installed the nightly build and this feature worked well. We were able to create links and open up Nagios alerts in Nagios that we had sourced into Elastic.

**Comment by torkelo** [comment]
that's great to hear, appreciate you testing it, good to know that it's solving that use case!

**Comment by your-eminence** [comment]
I've just tried this and works well.  Thankyou!  I did notice that setting the url causes the threshold colour to be ignored though.

**Comment by Krylanc3lo** [comment]
Hello Torkelo. Thanks a lot for this feature. it will be very useful.
However, I installed the nightly build but I am not able to make it work (I am using ElasticSearch data source). I added a link but I cannot click on the cell (nothing happens). Do you have any idea?

**Comment by torkelo** [comment]
Can you show your column style options? And the table? Does the column style refex match a column name?

**Comment by Krylanc3lo** [comment]
Thank you for your help, I do appreciate.

Here are my column style options:
![image](https://user-images.githubusercontent.com/22239771/29362156-bb39326a-8258-11e7-943f-9bbb408ee503.png)

and here are my table:
![image](https://user-images.githubusercontent.com/22239771/29362133-a782b7c8-8258-11e7-9a4c-9e3e1e0818c2.png)

I made sure my style matches a column name.

**Comment by inselbuch** [comment]
Try /source/ as the column name.



Frank Inselbuch
Unix Edge, Inc.
713-701-5421
inselbuch@unixedge.com<mailto:inselbuch@unixedge.com>
https://zoom.us/j/7137015421
Dial: +1 646 558 8656
Meeting ID: 713 701 5421

**Comment by Krylanc3lo** [comment]
Thank you Inselbuch, I tried but still nothing happens when I click on it :(

**Comment by torkelo** [comment]
you need to have the rule above the wildcard regex rule

**Comment by Krylanc3lo** [comment]
Hello,
It works perfectly, thanks a lot for your help :)

**Comment by taichatha** [comment]
> To make a column with link cells you add a column style and check the `Render value as link ` checkbox. That will make the link url & tooltip options appear.


<img width="997" alt="screen shot 2017-08-22 at 2 07 05 pm 2" src="https://user-images.githubusercontent.com/8629842/29580213-3c66181e-8743-11e7-9258-425817315296.png">

I don't have a checkbox that says 'Render Value as link'

**Comment by taichatha** [comment]
@torkelo I am not. I am on v4.4.

**Comment by antiquechrono** [comment]
Would it be possible to have a Column Style option for Dates where if you click on the date it sets the current dashboards time range to that date with a user configurable time before and after the date?This could help solve the #8310 feature request.

**Comment by inselbuch** [comment]
Okay, now how to I pass in the column name to populate a template variable? In other words, I have a display with a template variable $assetid. I would like to click on the table column and jump to the templated display with the template variable set to the value of the column.

**Comment by Ritesh-Sharma** [comment]
@torkelo : Hi, this is really a very useful feature,i was looking for it since far ; Could you please tell me how to get this nightly build and from where ? i tried seeking for Grafana 5,  at http://docs.grafana.org/builds/ but unable to find it.

**Comment by Ritesh-Sharma** [comment]
@torkelo : Hey , I got the correct link to download ..Sorry for trouble!!

**Comment by alexanderzobnin** [comment]
@inselbuch just use link to dashboard with `&var-assetid=$__cell`

**Comment by wfbrown53** [comment]
Lasted beta is 4.7.0 or is there a 5... somewhere?

**Comment by wfbrown53** [comment]
Latest is 4.7.0 beta or is there a 5... somewhere with this?

**Comment by inselbuch** [comment]
that feature was first available in 4.5.0-beta1
i believe it was released in 4.5.0 and has been available in every subsequent release

**Comment by wfbrown53** [comment]
Thank you sir. Can something similar be used in a graph panel? i.e. Use the Top 10 finds to populate a template dropdown?

**Comment by inselbuch** [comment]
I don’t understand your question.
But templating is quite limited, depending on your data source.
Good luck.

**Comment by munokar** [comment]
Hi,

first of all thanks for implementing that very nice feature!
I started using it and it really helps a lot.

I have some tables (time series aggregration) where the Colume 0 contains that alias of the metric.
This alias is created by aliasByNode using mutiple elements ( aliasByNode(3, 4, 6, 7) ).
Now I want to know whether there is an option to use these elements as individual elements for the URL since in some other dashboards I have these elements as template variables.

So a small example:
A dashboard shows as table the top 5 ports of a networking infrastructure where the alias is like network.switchname.slotname.portname. I want to add a drilldown link to the table line that goes to a dashboard showing the time series data as chart where the defined template variales are $Network, $Switch, $Slot and $Port.

So anyone an idea whether this is possible?

Thanks in advance

Timo

**Comment by kiplandiles** [comment]
I too really like this feature and everything seems to work as designed except for one issue. If I use a relative link and DO NOT click the "Open in new tab", nothing seems to change. Works great in a new tab. Have not tried it yet with a full URL.

Google Chrome Version 63.0.3239.132 (Official Build) (64-bit) on macOS High Sierra (10.13.2).

**Comment by malkia** [comment]
Same as @kiplandiles - In my case I have a top table showing all instances of applications that were run, then if I click on a column, I would like to display in table below the actual logs (actually I would like that 2nd table to hidden if the value is empty, but that's okay for now even like it)

**Comment by mankoffs** [comment]
Hi,
I have a question about the clickable cell links. I need to have several clickable links in the column cell. 
For example, in the cell will be the value: 'link1, link2'. Each of the links should be clickable with a specific url. Is it possible to implement it?

**Comment by inselbuch** [comment]
yes

Sent from my iPad

On Feb 2, 2018, at 9:43 AM, Sergio <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi,
I have a question about the clickable cell links. I need to have several clickable links in the column cell.
For example, in the cell will be the value: 'link1, link2'. Each of the links should be clickable with a specific url. Is it possible to implement it?

**Comment by mankoffs** [comment]
Do you have any examples?
At the current time, I see that this can be done through the sql function that would parse the string 'link1, link2, ..' to string: '<a href="https://link1">link1</a>, <a href="https://link2">link2</a>'.
but I'm not sure that grafana supports sql function in query.

**Comment by nsayedqt** [comment]
Hello everyone, I am still struggling to use this feature. I have 4 rows in my table, and 4 urls that i want each row to point to or linked to. I tried creating:
- 4 different rules, with each url and specifying the cell number. But since the column header name is same, it applies same url to all the rows.
- Also tried 1 rule with comma seperated urls with different cell numbers

Please help. Thanks.

![image](https://user-images.githubusercontent.com/23022014/38393450-7358a2b0-38f0-11e8-9bca-71f643e9216f.png)
![image](https://user-images.githubusercontent.com/23022014/38393533-b58a50a2-38f0-11e8-9143-101349c88175.png)

**Comment by inselbuch** [comment]
it may not be possible!



Frank Inselbuch
Unix Edge, Inc.
Fleet Surveillance and Analytics
713-701-5421
inselbuch@unixedge.com<mailto:inselbuch@unixedge.com>
https://zoom.us/j/7137015421

**Comment by dmitrios** [comment]
I don't think it is possible either. Perhaps, the workaround could be using simple service that handles link redirections with all the logic in it.

**Comment by nsayedqt** [comment]
Thanks for the replies, I thought i was using the feature incorrectly. Will try other work arounds. Thanks.

**Comment by Kenith** [comment]
@nsayedqt You could use the plugin [https://github.com/dmitrios/grafana-plugins-reftable](https://github.com/dmitrios/grafana-plugins-reftable) instead. I have same issue as you and the plugin works. If you have any questions about how to use it, feel free to ask me.

**Comment by byoungbnice** [comment]
I'm not able to edit the details of Link.  I have a number column called CoserverID and it shows as follows.  I'm using grafana 5.2.2.  Any suggestion on how I can get this to work would be greatly appreciated!

![reftable_link](https://user-images.githubusercontent.com/42948163/45001666-4d2ee000-af8c-11e8-8ed3-ca314c8e94da.png)

**Comment by kiplandiles** [comment]
I noticed that when I upgraded Grafana from 5.2.1 to 5.3.1 now my links in the table columns actually show an underline. While that makes it more obvious what columns are actually links it does use additional real estate and makes those columns a bit harder to read on a dense dashboard.

Is there a way to override the display of Links as underlines?

**Comment by torkelo** [comment]
No, there is no way to override it

**Comment by rfogel** [comment]
@malkia did you achieve what you were trying to do? I have the same problem.

Scenario: i have 2 tables, i need to click on a row on the first table and trigger an update on the second using some value from the row. I'm using Json datasource. Is that possible?

**Comment by Saipradeep37** [comment]
Can someone help me on how to redirect to new dashboard that has multiple rows and when we click on any of the rows it should redirect according to the data

**Comment by Great-Li-Xin** [comment]
I would definitely recommend this feature, suppose that we have the following table:

```
-------------------
server 1 | cpu 1
-------------------
server 2 | cpu 2
-------------------
```

I'd definitely appreciate that if I can directly click on the `CPU` col fo the second row then the link takes me to the corresponding `cpu 2 of server 2` page.

**Comment by tobiasdirksen** [comment]
Can't figure out how to filter a second panel by clicking on a first panel, both on the same dashboard. I have added a url like below to the column in the first panel, where I update the gateway variable when click on a cell. 
`http://servername/d/ogRQa9OWz/datahjemtagelse?orgId=1&refresh=1s&var-interval=1%20hour&var-gateway=$__cell`
However, the page doesn't refresh when clicking on the cells. If I refresh the page manually, it works fine, but that is not desirable.

**Comment by sabeDev** [comment]
> Can't figure out how to filter a second panel by clicking on a first panel, both on the same dashboard. I have added a url like below to the column in the first panel, where I update the gateway variable when click on a cell.
> `http://servername/d/ogRQa9OWz/datahjemtagelse?orgId=1&refresh=1s&var-interval=1%20hour&var-gateway=$__cell`
> However, the page doesn't refresh when clicking on the cells. If I refresh the page manually, it works fine, but that is not desirable.

Hi did you figure this, I cannot even seem to get the panel to display on refresh... it shows the correct the value in the hovered url (at the bottom) but never seems to populate from the select query; I am using WHERE whatever = '$variable'... where variable is e.g.:
var-test, would become WHERE whatever = '$test'
